cernopendata/api.py

```diff
@@ -0,0 +1,171 @@
+# -*- coding: utf-8 -*-
+"""API for manipulating file indices associated to a record."""
+
+import json
+from collections import OrderedDict
+
+from invenio_files_rest.models import (
+    Bucket,
+    BucketTag,
+    FileInstance,
+    ObjectVersion,
+    ObjectVersionTag,
+)
+from invenio_records_files.api import FileObject, FilesIterator, Record
+
+
+class FileIndexIterator(object):
+    """Class to iterate over the files."""
+
+    def __init__(self, record):
+        """Initialize iterator."""
+        self._it = None
+        self.record = record
+        self.model = record.model
+
+        self.file_indices = OrderedDict(
+            [(f["key"], f) for f in self.record.get("_file_indices", [])]
+        )
+
+    #    @property
+    #    def keys(self):
+    #        """Return file keys."""
+    #        return self.filesmap.keys()
+
+    #    def __len__(self):
+    #       """Get number of files."""
+    #        return ObjectVersion.get_by_bucket(self.bucket).count()
+
+    #    def __iter__(self):
+    #        """Get iterator."""
+    #        self._it = iter(sorted_files_from_bucket(self.bucket, self.keys))
+    #        return self
+
+    #    def next(self):
+    #        """Python 2.7 compatibility."""
+    #        return self.__next__()  # pragma: no cover
+
+    #    def __next__(self):
+    #        """Get next file item."""
+    #        obj = next(self._it)
+    #        return FileIndex(obj, self.filesmap.get(obj.key, {}))
+
+    #    def __contains__(self, key):
+    #        """Test if file exists."""
+    #        return (
+    #            ObjectVersion.get_by_bucket(self.bucket).filter_by(key=key).count()
+    #        )
+
+    #    def __getitem__(self, key):
+    #       """Get a specific file."""
+    #        obj = ObjectVersion.get(self.bucket, key)
+    #        if obj:
+    #            return FileIndex(obj, self.filesmap.get(obj.key, {}))
+    #        raise KeyError(key)
+
+    def flush(self):
+        """Flush changes to record."""
+        file_indices = self.dumps()
+        if file_indices or "_file_indices" in self.record:
```

Comment:
```suggestion
        if file_indices:
```

check if this is enough

Reply:
done